robotidy/transformers/SplitTooLongLine.py

```diff
@@ -0,0 +1,112 @@
+from itertools import tee
+
+from robot.api.parsing import (
+    ModelTransformer,
+    Token
+)
+
+
+def eol():
+    return Token(Token.EOL)
+
+
+class SplitTooLongLine(ModelTransformer):
+    def __init__(self, line_length: int = 120):
+        self.line_length = line_length
+
+    def visit_KeywordCall(self, node):  # noqa
+        if all(token.end_col_offset <= self.line_length for token in node.tokens[::-1]):
+            return node
+        if not node.get_tokens(Token.ARGUMENT):  # return if there are no arguments - nothing to split
+            return node
+        return self.split_keyword_call(node)
+
+    def split_keyword_call(self, node):
+        is_single_line = node.lineno == node.end_lineno
+
+        indent = node.tokens[0]
+        assignment = list(self.insert_seperator(node.get_tokens(Token.ASSIGN)))
+        keyword = node.get_token(Token.KEYWORD)
+
+        if is_single_line:
+            comment = node.get_tokens(Token.COMMENT)
+            comment_line = [indent] + comment + [eol()] if comment else []
+
+            head = comment_line + [indent] + assignment + [keyword]
+
+            tail = node.tokens[node.tokens.index(keyword) + 1:]
+            tail = list(filter(lambda t: t.type != Token.COMMENT, tail))
+
+            lines = []
+            for tokenized_line in self.pack_arguments(head, tail, indent):
+                lines.extend(tokenized_line)
+
+        else:
+            return node  # Not implemented yet
+
+        node.tokens = lines
+        return node
+
+    def pack_arguments(self, tokens_so_far, remaining_tokens, indent):
+        if not remaining_tokens:
+            return []
+
+        # Ensure we are in the correct place to pack
+        assert (not remaining_tokens or (
```

Comment:
assert are typically used in test, not in prod code. What will happen if assert will be false, the program will abruptly end?

Reply:
Asserts are normally used in production code too, and mypy even requires them in some places. You can optimize it out with `-O` flag to python interpretter. The idea is that it's better to end abruptly than to have a bug that is hard to find.

I will remove them later, I did it mostly to ensure that I'm having the right idea of what is going into this subroutine.